### DISCUSSION TITLE
long translations & sort by title

### DISCUSSION THREAD

**Discussion by ichnilatis-gr** [koreader/koreader#7201] [Q&A]
Two requests:

1) Sometimes there are translations of some options that are longer than the English original text. 
Could it be possible a pop-up window to be displayed with the full text of the option, after a long press?

![FileManager_2021-01-25](https://user-images.githubusercontent.com/76203123/105875585-a6d55080-6006-11eb-8b43-8c52b3f1a9ad.gif)

 2) Could be the option "sort by title" added? This would sort the documents by their title, according their metadata, and (if they haven't metadata) their filename. This solves the problem that, if the documents are sorted by name (that it doesn't always agree with their title in metadata)  they aren't displayed alphabetically.

Thank you!

**Comment by Frenzie** [comment]
In general, yeah, that's a good idea, but for this specifically perhaps it could be avoided by rewriting the localization slightly? Think "detailed list" → "details" and the like.

**Reply by ichnilatis-gr** [reply]
Does it pertain to my second request? Is it about an option in KOReader? Can you explain it to me?

Thank you!

**Reply by Frenzie** [reply]
No, it's about the first. The localization can and should take a lot of liberties with the text with an eye on various aspects (e.g., how natural/idiomatic does it sound, fitting in the allotted space, clarity). Sometimes the original should be changed too.

**Reply by ichnilatis-gr** [reply]
So, should I open two new issues about those requests?

**Reply by Frenzie** [reply]
But you do (most of) the Greek translation I believe, so that's an issue for yourself? ;-)

I'm trying to say that if you can satisfactorily rephrase it in Greek as something shorter along the lines of "Details with cover and metadata" or "Details with cover and info" then please do. That's standard localization practice.

Whether the original string should be changed is an almost completely independent question.

**Reply by ichnilatis-gr** [reply]
Unfortunately, this isn't always easy. Sometimes even a few words in Greek are longer than English... 
So, would it be difficult for a popup with full text to be created?

What about my second request?

**Reply by Frenzie** [reply]
> So, would it be difficult for a popup with full text to be created?

Possibly, it's hard to say without looking into it. Long press is used to give explanation and set defaults, so a default hold callback that simply shows the text could work. Difficulties here are primarily in the realm of design. It's much better to avoid disease than to cure it, and the proposed workaround feels like it might be a capitulation to bad design rather than an actual fix. Allowing quick hacks tend to come back to bite you a couple of years down the line.

Could you please file it as a bug?

> What about my second request?

I think that's already covered by https://github.com/koreader/koreader/issues/3864, so I'd suggest adding the use case there.

**Reply by ichnilatis-gr** [reply]
> Could you please file it as a bug?

How can I do this? However, if you think this would be a hassle, let's leave it.

**Reply by Frenzie** [reply]
Roughly what you did here, except in issues and following the template.

**Comment by poire-z** [comment]
> Could it be possible a pop-up window to be displayed with the full text of the option, after a long press?

 I agree it should be the translator job to try finding some short wording, and some wording that can still make sense if truncated.
We could have it displayed non-truncated on long-press, but we have long-press showing on some menu items a help_text. Having a menu _not_ react to long-press is an indication there it no additional help.
Having it always react would be a bit bothering, you'd just see the menu item title, that you'd have to eye on to know it's not some help text.
Having it only react when the text is truncated (not sure we can easily know that) could be a middle ground, dunno.

> Could be the option "sort by title" added? 

Quite complicated. Getting the title from the metadata means parsing the book.
We have the CoverBrowser plugin that may do that and cache it, but it's far away from the sorting code - and your current directory might not have had the metadata for all books extracted, so you could get partially bad results/sorting - or have to wait.
(Although it could be nice to delegate the sorting to this plugin, which could just do it via SQL to get all (cached data for) documents in the current directory, there's still the need for them to be cached...)
Really complicated.

**Reply by ichnilatis-gr** [reply]
O.K. Let's leave them.
I will try for shorter translations.
As far as it concerns my second request, I just have in mind the options that Pocketbook gives for sorting. After all, I can use this option in Pocketbook and open the files from there.

Thank you for your time!

**Reply by Frenzie** [reply]
Yes, I meant only if ..., not always.

**Reply by poire-z** [reply]
> Having it only react when the text is truncated (not sure we can easily know that) 

Looks like this might not be too complicated, I'll do it.

**Reply by ichnilatis-gr** [reply]
I noticed in the last version that you did it. It's nice. Thank you!
Nevertheless, I try to shorten the texts as much as I can...
